src/components/legend/draw.js

```diff
@@ -321,38 +320,35 @@ module.exports = function draw(gd) {
     }
 
     if(gd._context.editable) {
-        var xf,
-            yf,
-            x0,
-            y0,
-            lw,
-            lh;
+        var xf, yf, x0, y0;
+
+        legend.classed('cursor-move', true);
 
         dragElement.init({
             element: legend.node(),
             prepFn: function() {
-                x0 = Number(legend.attr('x'));
-                y0 = Number(legend.attr('y'));
-                lw = Number(legend.attr('width'));
-                lh = Number(legend.attr('height'));
-                setCursor(legend);
+                // regex pattern for 'translate(123.45px, 543.21px)'
+                var re = /(.*\()(\d*\.?\d*)([^\d]*)(\d*\.?\d*)([^\d]*)/,
+                    transform = legend.attr('transform')
+                    .replace(re, function(match, p1, p2, p3, p4) {
+                        return [p2, p4].join(' ');
+                    })
+                    .split(' ');
+
+                x0 = +transform[0] || 0;
+                y0 = +transform[1] || 0;
```

Comment:
@mdtusz can you make a `Lib` function actually? 

So that we test it in isolation (and re-use it eventually for range selectors and [update menus](https://github.com/plotly/streambed/issues/6064#issuecomment-214559755)).